pkg/apis/kubeone/v1beta1/defaults.go

```diff
@@ -127,8 +129,23 @@ func SetDefaults_ClusterNetwork(obj *KubeOneCluster) {
 		obj.ClusterNetwork.NodePortRange = DefaultNodePortRange
 	}
 	if obj.ClusterNetwork.CNI == nil {
+		canal := &CanalSpec{
+			MTU: "1450", // standard 1500 bytes - 50 VXLAN bytes
+		}
+
+		switch {
+		case obj.CloudProvider.AWS != nil:
+			canal.MTU = "8951" // 9001 AWS Jumbo Frame - 50 VXLAN bytes
+		case obj.CloudProvider.GCE != nil:
+			canal.MTU = "1410" // GCE specific 1460 bytes - 50 VXLAN bytes
+		case obj.CloudProvider.Hetzner != nil:
+			canal.MTU = "1400" // Hetzner specific 1450 bytes - 50 VXLAN bytes
+		case obj.CloudProvider.Openstack != nil:
+			canal.MTU = "1400" // Openstack specific 1450 bytes - 50 VXLAN bytes
+		}
```

Comment:
MTU should be defaulted regardless of is `clusterNetwork.CNI == nil`. We should skip defaulting only if `MTU != ""`.

Reply:
If I'll assign to obj.ClusterNetwork.CNI regardless of its nilness, that will override other possible options selected by the user (weave / external), isn't it?